experimental/trusted_shuffler/server/src/http.rs

```diff
@@ -26,26 +26,82 @@ use std::{
     time::Instant,
 };
 use tokio::time::Duration;
-use trusted_shuffler::{RequestHandler, TrustedShuffler};
-use trusted_shuffler_common::send_http_request;
+use trusted_shuffler::{
+    RequestHandler, TrustedShuffler, TrustedShufflerRequest, TrustedShufflerResponse,
+};
+use trusted_shuffler_common::send_with_request;
 
 struct HttpRequestHandler {
     backend_url: String,
 }
 
 #[async_trait]
 impl RequestHandler for HttpRequestHandler {
-    async fn handle(&self, request: Vec<u8>) -> anyhow::Result<Vec<u8>> {
-        let response = send_http_request(&self.backend_url, Method::POST, &request).await;
-        response.map_or_else(
-            |error| {
-                Err(anyhow!(
-                    "Couldn't receive response from the backend: {:?}",
-                    error
-                ))
-            },
-            Ok,
-        )
+    async fn handle(
+        &self,
+        request: TrustedShufflerRequest,
+    ) -> anyhow::Result<TrustedShufflerResponse> {
+        let mut request = trusted_to_hyper_request(request);
+
+        // We want to keep the path of the orginal request from the client.
+        let path = request.uri().path();
```

Comment:
Isn't this removing query params and potentially other parts of the URI? I think it should be possible to just clone the original URI, and then modify the "authority" part of that (which also avoids having to concatenate strings together). In particular, https://docs.rs/hyper/latest/hyper/struct.Uri.html#method.into_parts may be useful, together with https://docs.rs/http/0.2.8/http/uri/struct.Authority.html#method.from_static to create the new authority, and then replace it with https://docs.rs/hyper/latest/hyper/struct.Uri.html#method.from_parts .

Reply:
Done, thank you!

Reply:
Actually, I had to roll back, because it requires a static lifetime for backend_url, which would be a bigger change. I think we should in general store not a string, but an Authority in `HttpRequestHandler`. But I'd leave that for a seperate PR.